src/sentry/search/django/backend.py

```diff
@@ -136,12 +136,22 @@ def _build_queryset(self, project, query=None, status=None, tags=None,
             )
 
         if tags:
-            matches = self._tags_to_filter(project, tags)
-            if not matches:
-                return queryset.none()
-            queryset = queryset.filter(
-                id__in=matches,
-            )
+            filter_tags = {k: v for k, v in six.iteritems(tags) if k[0] != '-'}
+            exclude_tags = {k[1:]: v for k, v in six.iteritems(tags) if k[0] == '-'}
```

Comment:
Adding a special case. I think a query with both filter and exclude for the same tag is broken, therefor returning queryset.none() makes sense here.